functions/snippets/src/test/java/MockHttpResponse.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.google.cloud.functions.HttpResponse;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+class MockHttpResponse implements HttpResponse {
```

Comment:
Ok, so this class (and `MockHttpRequest`) should really also being using getters and setters and not have accessing public fields. 

That said, I spoke this over with Les this morning about best practices for mocking and he pointed out that my fears with mocking third-party interfaces were over-realized. I was under the impression you had to be careful to avoid mocking a third party interface because if it changed, you could mock a method that was removed. After researching a little more thoroughly it looks like that should throw a compile time error.  So it sounds like the best practice in this case would in fact be to mock the interface directly. 

So **TL;DR**, you were handling this right correctly the first time (and how you are handling it in a few other tests). Sorry for making you spend time on my waffling. Please feel free to revert the effected tests to your earlier commit.

Reply:
Thanks for clarifying! Will revert.